File: Python/Simulations/CreateSIPDataframe.py
# Load packages
import pandas as pd
import numpy as np

def CreateSIPDataframe(name_of_items,
                       list_of_items,
                       number_of_trials=10000):
    # Create dataframe of trials
    df_sips = pd.DataFrame(columns=[
        'Trial',
        name_of_items
    ])
    
    # Iterate through list of items
    for item in list(set(list_of_items)):
        # Create trial column with item name
        df_temp = pd.DataFrame({
            'Trial': np.linspace(1, number_of_trials, number_of_trials, dtype='int'),
            name_of_items: item
        })
        # Append to SIP dataframe
        df_sips = df_sips.append(df_temp)
        
    # Return SIP DataFrame
    return(df_sips)

File: R/Simulations/CreateSIPDataframe.R
# Load packages
library(dplyr)

# Declare function
CreateSIPDataframe = function(name_of_items,
                              list_of_items,
                              number_of_trials=10000) {
  # Create place holder dataframe
  df_sip = data.frame()
  
  # Keep only unique set of items
  list_of_items = unique(list_of_items)
  
  # Iterate through list of items
  for (i in 1:length(list_of_items)) {
    # Create dataframe of rows for each trial
    df_temp = data.frame(1:number_of_trials)
    colnames(df_temp) = "Trial"
    
    # Add item as column
    df_temp[[name_of_items]] = list_of_items[i]
    
    # Concatenate item and trial number as key
    row.names(df_temp) = paste0(df_temp[["Trial"]], "-", df_temp[[name_of_items]])
    
    # Append to placeholder dataframe
    df_sip = bind_rows(df_sip, df_temp)
    rm(df_temp)
  }
  
  # Return dataframe
  return(df_sip)
}
